cognite/client/data_classes/data_modeling/filters.py

```diff
@@ -0,0 +1,312 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Optional, Sequence, Tuple, Union, cast, final
+
+ContainerReference = Tuple[str, str]
+ViewReference = Union[Tuple[str, str], Tuple[str, str, str]]
+
+PropertyReference = Union[Tuple[str, ...], List[str]]
+
+RawValue = Union[str, float, bool, Sequence, Mapping[str, Any]]
+
+
+@dataclass
+class PropertyReferenceValue:
+    property: PropertyReference
+
+
+@dataclass
+class ParameterValue:
+    parameter: str
+
+
+FilterValue = Union[RawValue, PropertyReferenceValue, ParameterValue]
+FilterValueList = Union[Sequence[RawValue], PropertyReferenceValue, ParameterValue]
+
+
+def _dump_filter_value(filter_value: FilterValueList | FilterValue) -> Any:
+    if isinstance(filter_value, PropertyReferenceValue):
+        return {"property": filter_value.property}
+    if isinstance(filter_value, ParameterValue):
+        return {"parameter": filter_value.parameter}
+    else:
+        return filter_value
+
+
+def _load_filter_value(value: Any) -> FilterValue | FilterValueList:
+    if isinstance(value, Mapping) and len(value.keys()) == 1:
+        value_key = next(iter(value.keys()))
```

Comment:
Consider using unpacking instead:
```suggestion
        value_key, = value
```

Reply:
Elegant.